mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/memory/VaultGarbageCollectionTask.java

```diff
@@ -0,0 +1,115 @@
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault.memory;
+
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+import org.apache.james.vault.search.CriterionFactory;
+import org.apache.james.vault.search.Query;
+
+public class VaultGarbageCollectionTask implements Task {
+    public static class AdditionalInformation implements TaskExecutionDetails.AdditionalInformation {
+        private final ZonedDateTime beginningOfRetentionPeriod;
+        private final long handledUserCount;
+        private final long permanantlyDeletedMessages;
+        private final long vaultSearchErrorCount;
+        private final long deletionErrorCount;
+
+        AdditionalInformation(ZonedDateTime beginningOfRetentionPeriod, long handledUserCount, long permanantlyDeletedMessages, long vaultSearchErrorCount, long deletionErrorCount) {
+            this.beginningOfRetentionPeriod = beginningOfRetentionPeriod;
+            this.handledUserCount = handledUserCount;
+            this.permanantlyDeletedMessages = permanantlyDeletedMessages;
+            this.vaultSearchErrorCount = vaultSearchErrorCount;
+            this.deletionErrorCount = deletionErrorCount;
+        }
+
+        public  ZonedDateTime getBeginningOfRetentionPeriod() {
+            return beginningOfRetentionPeriod;
+        }
+
+        public long getHandledUserCount() {
+            return handledUserCount;
+        }
+
+        public long getPermanantlyDeletedMessages() {
+            return permanantlyDeletedMessages;
+        }
+
+        public long getVaultSearchErrorCount() {
+            return vaultSearchErrorCount;
+        }
+
+        public long getDeletionErrorCount() {
+            return deletionErrorCount;
+        }
+    }
+
+    private static final String TYPE = "deletedMessageVaultRetention";
```

Comment:
can we use the same pattern of TYPE for tasks relate to Vault? I did `deletedMessages/restore` and `deletedMessages/export` for restore and export.

Reply:
`deletedMessages/garbageCollection`